uwblueprint/childrens-aid-society
# create initial logic for cases context

## Notion ticket link

[Create cases context to use throughout app](https://www.notion.so/uwblueprintexecs/134e1f159caa4f73939d11a700c325ad?v=39cf7d5a98384eb9af2bff2265d0ac8a&p=d61821d58d3546adb1510fcdf6f29686&pm=s)



## Implementation description
* As we handle case data and pass it around into different components and pages, using Context will simplify and make it a lot easier to retrieve and update the cases in the app. 
* Implementation logic is pretty standard for using context and reducers, similar to SampleContext with a default context, dispatcher, and reducer.

* We'll need to build this out more: adding actions for the reducer, editing some of the existing states/props passed (ex. cases in HomePage) to use context instead of hard coding

here's some react docs for context (haha) about the idea of context and reducers if you aren't familiar with it: https://react.dev/reference/react/useContext
https://react.dev/reference/react/useReducer


## Steps to test
1. nothing to really test right now, we can edit the DEFAULT_CASES and then log it to see it in the HomePage.



## What should reviewers focus on?
* 


## Checklist
